### DISCUSSION TITLE
How to handle websocket disconnect from client

### DISCUSSION THREAD

**Discussion by MichalPham** [fastapi/fastapi#11298] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def ws_receiver(websocket: WebSocket) -> None:
    """Receive messages from client, we dont care about them except disconnect message."""
    async for _ in websocket.iter_text():
        pass

@app.websocket("/{item_id}/")
async def auction_detail_ws(
    websocket: WebSocket,
    item_id: UUID4 = Path(title="UUID of retrieved item"),
    broadcaster: AuctionBroadcast = Depends(get_broadcaster),
):
    """Subscribe user to channel and send back published messages."""
    await websocket.accept()
    async with anyio.create_task_group() as task_group:
        # run until first is complete, wait for disconnect event
        async def run_ws_receiver() -> None:
            try:
                await ws_receiver(websocket=websocket)
            finally:
                task_group.cancel_scope.cancel()

        task_group.start_soon(run_ws_receiver)
        await broadcaster_sender(websocket, broadcaster, str(auction_id))
    print("gggggg")
```

### Description

When disconectiong from client side (httpx-ws, js) using their respective `close` methods, the `startlette.websocket.receive_(text|bytes)` fails with `RuntimeError` and not with `WebSocketDisconnect`. Its caused by this [line](https://github.com/encode/starlette/blob/master/starlette/websockets.py#L51) because the message type is `websocket.close` and not `websocket.disconnect`. I have not found a way how to send `websocket.disconnect` looking at the implementation of starlette [close method](https://github.com/encode/starlette/blob/master/starlette/websockets.py#L202) it also sends `websocket.close`

So the question really is how do i correctly disconect from client side.
### Operating System

Linux

### FastAPI Version

0.109.0

### Pydantic Version

2.6.0

### Python Version

Python 3.11

**Comment by YuriiMotov** [comment]
Have you solved this?
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
